src/net/RTP/MediaStreamTrack.cs

```diff
@@ -262,5 +263,23 @@ public bool IsSsrcMatch(uint ssrc)
         {
             return Capabilities?.FirstOrDefault(x => x.ID == payloadID);
         }
+
+        /// <summary>
+        /// Returns the next SeqNum to be used in the RTP Sequence Number header field for media packets
+        /// sent using this media stream. 
+        /// </summary>
+        /// <returns></returns>
+        public ushort GetNextSeqNum()
+        {
+            var actualSeqNum = m_seqNum;
+            int expectedSeqNum;
+            do
+            {
+                expectedSeqNum = actualSeqNum;
+                int nextSeqNum = (actualSeqNum >= UInt16.MaxValue) ? (ushort)0 : (ushort)(actualSeqNum + 1);
+                actualSeqNum = Interlocked.CompareExchange(ref m_seqNum, nextSeqNum, expectedSeqNum);
+            } while (expectedSeqNum != actualSeqNum); // Try as long as compare-exchange was not successful; in most cases, only one iteration should be needed

```

Comment:
It would be nice to have a safety valve here to avoid any possibility of an infinite loop. I usually use something like an "attempts" variable and if it reaches 10 exit and throw an exception.

Reply:
Altough I don't think the safety valve will ever trigger(or if it does, something else is most likely a problem), it should not hurt performance, because this is only a local variable.